core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -96,23 +97,37 @@ public RowDelta validateNoConflictingDeleteFiles() {
   }
 
   @Override
-  protected void validate(TableMetadata base, Snapshot snapshot) {
-    if (base.currentSnapshot() != null) {
+  public RowDelta toBranch(String branch) {
+    targetBranch(branch);
+    return this;
+  }
+
+  @Override
+  protected void validate(TableMetadata base, Snapshot parent) {
+    if (parent != null) {
+      if (startingSnapshotId != null) {
+        Preconditions.checkArgument(
+            SnapshotUtil.isAncestorOf(parent.snapshotId(), startingSnapshotId, base::snapshot),
+            "Snapshot %s is not an ancestor of %s",
```

Comment:
I think a bit more context would be helpful if you ever encounter this error. `"Cannot validate changes: starting snapshot %s is not an ancestor of parent snapshot %s (branch %s)"`

Reply:
@amogh-jahagirdar Is this even required anymore ? We are anyway doing ancestor check in https://github.com/apache/iceberg/blob/a1d994c131dc5b5f42e1693e92e058493382593e/core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java#L742